src/web/releases.rs

```diff
@@ -497,6 +497,69 @@ impl Default for Search {
     }
 }
 
+fn redirect_to_random_crate(req: &Request, conn: &mut PoolClient) -> IronResult<Response> {
+    // since there is a chance of this query returning an empty result,
+    // we retry a certain amount of times, and log a warning.
+    for _ in 0..20 {
+        let rows = ctry!(
+            req,
+            conn.query(
+                "WITH params AS (
+                    -- get maximum possible id-value in crates-table
+                    SELECT last_value AS max_id FROM crates_id_seq
+                )
+                SELECT
+                    crates.name,
+                    releases.version,
+                    releases.target_name
+                FROM (
+                    -- generate 500 random numbers in the ID-range. 
+                    -- this might have to be increased when we have to repeat 
+                    -- this query too often. 
+                    -- it depends on the percentage of crates with > 100 stars
+                    SELECT DISTINCT 1 + trunc(random() * params.max_id)::INTEGER AS id
+                    FROM params, generate_series(1, 500)
+                ) AS r
+                INNER JOIN crates ON r.id = crates.id
+                INNER JOIN releases ON crates.latest_version_id = releases.id
+                INNER JOIN github_repos ON releases.github_repo = github_repos.id
+                WHERE
+                    releases.rustdoc_status = TRUE AND
+                    github_repos.stars >= 100
+                LIMIT 1
+                ",
+                &[]
+            ),
+        );
+
+        if let Some(row) = rows.into_iter().next() {
+            let name: String = row.get("name");
+            let version: String = row.get("version");
+            let target_name: String = row.get("target_name");
+            let url = ctry!(
+                req,
+                Url::parse(&format!(
+                    "{}/{}/{}/{}",
+                    redirect_base(req),
+                    name,
+                    version,
+                    target_name
+                )),
+            );
+
+            let mut resp = Response::with((status::Found, Redirect(url)));
+            resp.headers.set(Expires(HttpDate(time::now())));
+
+            return Ok(resp);
```

Comment:
```suggestion
            log::debug!("finished random crate search on iteration {}", i);
            return Ok(resp);
```
Reporting how many tries each query took seems useful, you'll have to add `i` as the loop counter too